BoundedBlockingQueue.h moving a local object in a return statement prevents copy elision

distro
```
Ubuntu 19.10 Desktop
```

gcc
```
9.2.1
```

cmake
```
3.13.4
```

libboost-dev
```
1.67.0.2
```

muduo
```
master branch HEAD 0edc4c
```

**BUILD ERROR**
```
++ pwd
+ SOURCE_DIR=/home/zhyncs/Documents/muduo
+ BUILD_DIR=../build
+ BUILD_TYPE=release
+ INSTALL_DIR=../release-install-cpp11
+ CXX=g++
+ ln -sf ../build/release-cpp11/compile_commands.json
+ mkdir -p ../build/release-cpp11
+ cd ../build/release-cpp11
+ cmake -DCMAKE_BUILD_TYPE=release -DCMAKE_INSTALL_PREFIX=../release-install-cpp11 -DCMAKE_EXPORT_COMPILE_COMMANDS=ON /home/zhyncs/Documents/muduo
-- The C compiler identification is GNU 9.2.1
-- The CXX compiler identification is GNU 9.2.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Boost version: 1.67.0
-- Could NOT find Protobuf (missing: Protobuf_INCLUDE_DIR) 
-- Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR) 
-- Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR) 
-- CXX_FLAGS = -g -DCHECK_PTHREAD_RETURN_VALUE -D_FILE_OFFSET_BITS=64 -Wall -Wextra -Werror -Wconversion -Wno-unused-parameter -Wold-style-cast -Woverloaded-virtual -Wpointer-arith -Wshadow -Wwrite-strings -march=native -std=c++11 -rdynamic -O2 -DNDEBUG
-- Looking for accept4
-- Looking for accept4 - found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/zhyncs/Documents/build/release-cpp11
+ make
Scanning dependencies of target muduo_base
[  1%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/AsyncLogging.cc.o
[  1%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/Condition.cc.o
[  1%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/CountDownLatch.cc.o
[  1%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/CurrentThread.cc.o
[  2%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/Date.cc.o
[  2%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/Exception.cc.o
[  2%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/FileUtil.cc.o
[  3%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/LogFile.cc.o
[  3%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/Logging.cc.o
[  3%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/LogStream.cc.o
[  3%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/ProcessInfo.cc.o
[  5%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/Timestamp.cc.o
[  5%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/Thread.cc.o
[  5%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/ThreadPool.cc.o
[  6%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/TimeZone.cc.o
[  6%] Linking CXX static library ../../lib/libmuduo_base.a
[  6%] Built target muduo_base
Scanning dependencies of target logging_test
[  6%] Building CXX object muduo/base/tests/CMakeFiles/logging_test.dir/Logging_test.cc.o
[  6%] Linking CXX executable ../../../bin/logging_test
[  6%] Built target logging_test
Scanning dependencies of target logfile_test
[  6%] Building CXX object muduo/base/tests/CMakeFiles/logfile_test.dir/LogFile_test.cc.o
[  7%] Linking CXX executable ../../../bin/logfile_test
[  7%] Built target logfile_test
Scanning dependencies of target processinfo_test
[  7%] Building CXX object muduo/base/tests/CMakeFiles/processinfo_test.dir/ProcessInfo_test.cc.o
[  8%] Linking CXX executable ../../../bin/processinfo_test
[  8%] Built target processinfo_test
Scanning dependencies of target date_unittest
[  8%] Building CXX object muduo/base/tests/CMakeFiles/date_unittest.dir/Date_unittest.cc.o
[  8%] Linking CXX executable ../../../bin/date_unittest
[  8%] Built target date_unittest
Scanning dependencies of target boundedblockingqueue_test
[ 10%] Building CXX object muduo/base/tests/CMakeFiles/boundedblockingqueue_test.dir/BoundedBlockingQueue_test.cc.o
In file included from /home/zhyncs/Documents/muduo/muduo/base/tests/BoundedBlockingQueue_test.cc:1:
/home/zhyncs/Documents/muduo/muduo/base/BoundedBlockingQueue.h: In instantiation of ‘T muduo::BoundedBlockingQueue<T>::take() [with T = std::__cxx11::basic_string<char>]’:
/home/zhyncs/Documents/muduo/muduo/base/tests/BoundedBlockingQueue_test.cc:71:33:   required from here
/home/zhyncs/Documents/muduo/muduo/base/BoundedBlockingQueue.h:65:27: error: moving a local object in a return statement prevents copy elision [-Werror=pessimizing-move]
   65 |     return std::move(front);
      |                           ^
/home/zhyncs/Documents/muduo/muduo/base/BoundedBlockingQueue.h:65:27: note: remove ‘std::move’ call
/home/zhyncs/Documents/muduo/muduo/base/BoundedBlockingQueue.h: In instantiation of ‘T muduo::BoundedBlockingQueue<T>::take() [with T = std::unique_ptr<int>]’:
/home/zhyncs/Documents/muduo/muduo/base/tests/BoundedBlockingQueue_test.cc:91:39:   required from here
/home/zhyncs/Documents/muduo/muduo/base/BoundedBlockingQueue.h:65:27: error: moving a local object in a return statement prevents copy elision [-Werror=pessimizing-move]
/home/zhyncs/Documents/muduo/muduo/base/BoundedBlockingQueue.h:65:27: note: remove ‘std::move’ call
cc1plus: all warnings being treated as errors
make[2]: *** [muduo/base/tests/CMakeFiles/boundedblockingqueue_test.dir/build.make:63: muduo/base/tests/CMakeFiles/boundedblockingqueue_test.dir/BoundedBlockingQueue_test.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:340: muduo/base/tests/CMakeFiles/boundedblockingqueue_test.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

--- a/muduo/base/BlockingQueue.h
+++ b/muduo/base/BlockingQueue.h
@@ -52,7 +52,7 @@ class BlockingQueue : noncopyable
     assert(!queue_.empty());
     T front(std::move(queue_.front()));
     queue_.pop_front();
-    return std::move(front);
+    return front;
   }
 
   size_t size() const

--- a/muduo/base/BoundedBlockingQueue.h
+++ b/muduo/base/BoundedBlockingQueue.h
@@ -62,7 +62,7 @@ class BoundedBlockingQueue : noncopyable
     T front(std::move(queue_.front()));
     queue_.pop_front();
     notFull_.notify();
-    return std::move(front);
+    return front;
   }
 
   bool empty() const

